Add 3-month maintenance windows and groups

gdelisle opened the pull request:
Weekly is too frequently to patch and reboot most servers, so this PR adds maintenance windows and groups that will run every 3 months. These will run on the second Tuesday, Wednesday, and Thursday of every third month (January, April, July, October). These dates don't conflict with any holidays.

gdelisle committed:
Add 3-month maintenance windows and groups

gdelisle committed:
Case changes

paul-e-allen commented:
Hi @gdelisle: So the idea with these edits is to leave the weekly agent installs/updates, and weekly patch scans, but to only do the patching once every three months? Do you use the weekly patching at all (e.g., 24x7a)?

I was originally thinking that you could accomplish something similar by adding a new schedule block in the `MapEnvironmentParams`
```
    ThreeMonthCronSchedule:
      24x7a: 'cron(0 0 7 ? */3 TUE#2 *)'
      24x7b: 'cron(0 0 7 ? */3 WED#2 *)'
      24x7c: 'cron(0 0 7 ? */3 THU#2 *)'
      9x5default: 'cron(0 15 10 ? * MON *)'
      weeklytasks: 'cron(0 0 10 ? * MON *)'
      weeklyscans: 'cron(0 0 11 ? * MON *)'
```

Does that approach accomplish your goals?

gdelisle commented:
@paul-e-allen It accomplishes my goals. But by leaving both weekly and three-month schedules in place at the same time, it allows for a make-up window. For example, I've noticed that occasionally patching will fail on an instance that lacks enough disk space to download the updates. In that case, it is nice to have a weekly window to move the instance into, so it can be patched sooner than three months from now. But would that also be achievable by changing its tag to 9x5default which remains weekly?



On July 23, 2019 at 11:06:47 AM, paul-e-allen (notifications@github.com<mailto:notifications@github.com>) wrote:

Hi @gdelisle<https://github.com/gdelisle>: So the idea with these edits is to leave the weekly agent installs/updates, and weekly patch scans, but to only do the patching once every three months? Do you use the weekly patching at all (e.g., 24x7a)?

I was originally thinking that you could accomplish something similar by adding a new schedule block in the MapEnvironmentParams

    ThreeMonthCronSchedule:
      24x7a: 'cron(0 0 7 ? */3 TUE#2 *)'
      24x7b: 'cron(0 0 7 ? */3 WED#2 *)'
      24x7c: 'cron(0 0 7 ? */3 THU#2 *)'
      9x5default: 'cron(0 15 10 ? * MON *)'
      weeklytasks: 'cron(0 0 10 ? * MON *)'
      weeklyscans: 'cron(0 0 11 ? * MON *)'


Does that approach accomplish your goals?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/CU-CommunityApps/ct-patching-scanning/pull/2?email_source=notifications&email_token=AIROZU77VVGLTMF5FAQV7KLQA4NAJA5CNFSM4IBMK322YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD2TNVUI#issuecomment-514251473>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AIROZUYJNIII7POUPTIG7P3QA4NAJANCNFSM4IBMK32Q>.

gdelisle committed:
Redo three-month patching

Removed the 3-month tag windows and inserted 3-month cron schedule, as per Paul's comment. There should be little need for both weekly and 3-month windows to be used, and 9x5 can be used for weekly tasks.

gdelisle commented:
I've changed my PR to implement what you suggested. I agree that it's a cleaner approach and meets my needs. I've loaded this template into our sandbox account but I won't go live with it without your approval.
